### ISSUE TITLE
name[casing] issue should be raised for "notify" field under a task

### ISSUE THREAD

**Issue by audgirka** [ansible/ansible-lint#4035]
When we raise a `name[casing]` issue for a task under handler, the same should be raised for a task which is calling this handler task using `notify` task parameter

Related #4028

**Comment by pneerincx** [comment]
This produces false errors for cases where a handler uses an explicit `listen`, which does not need to start with an uppercase character. Hence ```notify``` does not need to start with an uppercase character in that case either. 

E.g. this handler:

```
- name: This handler name started with an uppercase character
   ansible.builtin.command:
     cmd: echo "Handler executed"
  changed_when: true
  listen: execute-my-handler
```

in combination with this task:
```
- name: This task name started with an uppercase character
   ansible.builtin.debug:
     cmd: echo "Just a dummy task to notify a handler."
  changed_when: true
  execute: execute-my-handler
```
will trigger
```
name[casing]: Task notify 'execute-my-handler' should start with an uppercase letter.
```
even though both task and handler names start with uppercase characters.

**Comment by gardar** [comment]
I don't like this one.

I usually stick to simple keywords when notifying handlers which I define under `listen`. And the name of the handler I use as a description of the task.

Checking the casting of the handler task name should be enough, it should catch any task names. No need to do it for the `listen`.
